### PR TITLE
Parse trait specs.

### LINKED ISSUE #179: Parsing of Prusti attributes on methods
Since the 2018->2020 compiler upgrade, `prusti-spec` doesn't support `requires` and `ensures` attributes on methods in traits and in type implementations. After #177, the same happens for `pure` and `trusted`.

### PR DESCRIPTION
**Pull request by vakaras**
A work in progress to fix #179

### CODE REVIEW

**Comment by fpoli** [comment]
Why the `verify-test` command? If it's for debugging we could put scripts like that in a `scripts` folder, to keep `x.py` lean. I also have some scripts which I could add, for example to move passing tests from `tests_old` to `tests`.

**Comment by vakaras** [comment]
> Why the `verify-test` command? If it's for debugging we could put scripts like that in a `scripts` folder

`./x.py test <test>` prints error messages in json, which is very hard to read. So, running a specific test in a “normal” way seems to be a command that one would use often. Especially, if it also configures the necessary flags correctly like checking for overflows (not done yet).

> to keep `x.py` lean.

I would like to have `x.py` as a main entry point because it makes much easier to discover commands. If the file grows too big, we can just split into multiple Python modules.

> I also have some scripts which I could add, for example to move passing tests from `tests_old` to `tests`.

I think it would be useful if you shared them. We can delete them after we have finished the migration completely.

**Comment by fpoli** [comment]
> `./x.py test <test>` prints error messages in json, which is very hard to read. So, running a specific test in a “normal” way seems to be a command that one would use often. Especially, if it also configures the necessary flags correctly like checking for overflows (not done yet).

I see the plan, the configuration flags would be useful.

> I think it would be useful if you shared them

Here: https://gist.github.com/fpoli/291f2a1dbabe33f0b0caed380123f69d. I'll add it in the next PR.

File: prusti-viper/src/encoder/procedure_encoder.rs [hunk 1]
```diff
@@ -2064,6 +2076,7 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
                 args,
                 destination,
                 called_def_id,
+                None, // FIXME: This is almost definitely wrong.
```
Comment (Aurel300) [inline prusti-viper/src/encoder/procedure_encoder.rs:2079]: I think it would be good to mention why?

**Pull request merged**